libraries/lib-snapping/SnapUtils.cpp

```diff
@@ -78,8 +78,34 @@ Identifier ReadSnapTo()
 {
    if (gPrefs->HasEntry(SnapToKey))
       return SnapToSetting.Read();
+   
+   static const std::unordered_map<wxString, wxString> selectionFormatLookup = {
+      { L"seconds", L"seconds" },
+      { L"seconds + milliseconds", L"milliseconds" },
+      { L"hh:mm:ss", L"seconds" },
+      { L"dd:hh:mm:ss", L"seconds" },
+      { L"hh:mm:ss + hundredths", L"centiseconds" },
+      { L"hh:mm:ss + milliseconds", L"milliseconds" },
+      { L"hh:mm:ss + samples", L"samples" },
+      { L"samples", L"samples" },
+      { L"hh:mm:ss + film frames (24 fps)", L"film_24_fps" },
+      { L"film frames (24 fps)", L"film_24_fps" },
+      { L"hh:mm:ss + NTSC drop frames", L"ntsc_29.97_fps" },
+      // Well, not really. Snapping with the "bent" time sounds funny anyway.
+      { L"hh:mm:ss + NTSC non-drop frames", L"ntsc_30_fps" },
+      { L"NTSC frames", L"ntsc_29.97_fps" },
+      { L"hh:mm:ss + PAL frames (25 fps)", L"film_25_fps" },
+      { L"PAL frames (25 fps)", L"film_25_fps" },
+      { L"hh:mm:ss + CDDA frames (75 fps)", L"cd_75_fps" },
+      { L"CDDA frames (75 fps)", L"cd_75_fps" },
+   };
```

Comment:
Essentially this is a many-to-one mapping from identifiers of selection formats, to identifiers of their grain size, and the latter identifiers are then mapped to snap functors by another registry.

This repetition of lots of string literals that are private to ParsedNumericConverterFormatter.cpp doesn't feel good.

Let it stand for now, but now I think the registry of snap functors wants to be placed in the same library with the registry of NumericConverterItems, and that registry should implement this map with extra information in the registered items, which are then accessed from SnapUtils with a Visit.

Reply:
This is only needed for config migration, it is not used otherwise. Please, let's not overcomplicate this. I would rather remove it and stick to default.

Reply:
I really want to stress that again - I don't think that snapping and numeric formatters are related. They are separate concepts and snapping must not be handled by lib-numeric-format.

Reply:
I disagree — I think it is reasonable that some form of information about the unit could be part of the registry of formats and accessible to the visitor.

Reply:
So do I. Formatting time to string is unrelated to snapping, they must not share libraries. What happens now is just a configuration migration. It is safe (that's it - a failure has no user visible implications), it is not used anywhere in calculations. It will not be used anywhere in calculations.

If you are uncomfortable with it - the only fix is to remove it. I have very strong opinion about it. Half of this PR is about breaking the ties between snapping and time display. 

Under absolutely no condition snapping should rely on numeric format functions. What you see in this map is a simple way of converting the list of well known values from the old Audacity versions to the new one. It does not introduce any kind of dependency between libraries.

Reply:
Ok there is agreement about something.

I said “let it stand” and still think that.

There are more important structural issues.

But this change made me think that:

Some notion of “identifier of time unit” is needed

These identifiers should not be a closed-ended list all in one place and each string literal should occur only once in the source

The registry of formats includes one such identifier in each leaf, which a Visitor can see, so that repetition of that registry’s key strings is not needed

The registry of snap functions is keyed on the same

These criteria are not met yet in the PR

Reply:
Maybe I’m arguing that this bit of logic needs to be elevated to a higher level library.  Maybe lib-snapping holds the registry of snap functions but SnapUtils, which finds and applies the functions, goes up to lib-time-frequency-selection.

Reply:
Maybe the correction of the Setting object at startup time, migrating and less config, is the only thing to elevate to a higher library.  Then snapping and formats depend neither on the other?

Reply:
Why? What complex logic do you expect from "finding" snapping functions? Unlike NumericConverterType it is not set anywhere explicitly in code.

Reply:
I have pushed a commit trying to clarify the intention of the `ReadSnapTo`. If you are not satisfied and still consider it to be dependency - I will remove that code. Resetting snap value to second would be just a minor inconvenience for the users.